blueocean-events/src/main/java/io/jenkins/blueocean/events/PipelineEventListener.java

```diff
@@ -71,8 +71,12 @@ public void onNewHead(FlowNode flowNode) {
         // test whether we have a stage node
         if (PipelineNodeUtil.isStage(flowNode)) {
             List<String> branch = getBranch(flowNode);
-            currentStageName.put(flowNode.getExecution(), flowNode.getDisplayName());
-            currentStageId.put(flowNode.getExecution(), flowNode.getId());
+            if(flowNode.getExecution()!=null && flowNode.getDisplayName()!=null) {
```

Comment:
My only question about this is what scenarios is `flowNode.getExecution()` ever `null`? Just curious if it's indicative of a bigger/different problem.

Reply:
well sure flowNode.getExecution() shouldn't be null. I will remove but this doesn't hurt neither have a big impact....